docs/components/clerk-provider.mdx

```diff
@@ -1,47 +1,35 @@
 ---
 title: '`<ClerkProvider>`'
-description: The <ClerkProvider> component wraps your React application to provide active session and user context to Clerk's hooks and other components.
+description: The <ClerkProvider> component provides session and user context to Clerk's hooks and components.
 ---
 
-The `<ClerkProvider>` component wraps your React application to provide active session and user context to Clerk's hooks and other components.
+The `<ClerkProvider>` component is required to integrate Clerk into your React application, providing session and user context to Clerk's hooks and components.
+
+The recommended approach is to wrap your entire app with `<ClerkProvider>` at the entry point to make authentication globally accessible. If you only need authentication for specific routes or pieces of your application, render `<ClerkProvider>` deeper in the component tree. This allows you to implement Clerk's functionality precisely where required without impacting the rest of your app.
 
 ## Usage
 
 The `<ClerkProvider>` component must be added to your React entrypoint.
 
 <Tabs items={["Next.js", "React"]}>
   <Tab>
-    <CodeBlockTabs options={["App Router", "Pages Router"]}>
```

Comment:
Our rule of thumb for Pages Router examples:
Remove unnecessary Pages Router examples for Next.js example. An example is considered unnecessary if it has the same exact code as an App Router example. In contrast, a specific example of being necessary is when an App Router example uses auth() and an Pages Router example uses getAuth(). These can be argued for, as the two different router types require two different methods for retrieving auth information.

for this case, the code is not exactly the same - the layout for an app router, and the `_app.tsx` file for pages router are pretty different. I would keep both.